src/Microsoft.AspNet.OData.Shared/TypeHelper.cs

```diff
@@ -574,7 +574,7 @@ internal static PropertyInfo GetProperty(Type targetType, string propertyName)
         /// <returns>true if the  type has a default constructor; otherwise returns false.</returns>
         internal static bool HasDefaultConstructor(Type type)
         {
-            return (!type.IsClass) || (!type.IsAbstract) || type.GetConstructors(BindingFlags.Public | BindingFlags.Instance)
+            return (!type.IsClass) || type.GetConstructors(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static) 
```

Comment:
Which use cases does BindingFlags.NonPublic cover?
BindingFlags.NonPublic will cause private, protected, internal and protected internal constructors to be returned and I'm not sure we can invoke any of those if the class exists in a separate library